pygmt/src/magnetic_rose.py

```diff
@@ -0,0 +1,119 @@
+from pygmt.params import Box, Position
+
+
+def magnetic_rose(  # noqa: PLR0913
+    self,
+    position: Position | None = None,
+    width: float | str | None = None,
+    labels: Sequence[str] | bool = False,
+    outer_pen: str | bool = False,
+    inner_pen: str | bool = False,
+    declination: float | None = None,
+    declination_label: str | None = None,
+    intervals: Sequence[float] | None = None,
+    box: Box | bool = False,
+    perspective: str | bool = False,
+    verbose: Literal["quiet", "error", "warning", "timing", "info", "compat", "debug"]
+    | bool = False,
+    transparency: float | None = None,
+):
+    """
+    Add a magnetic rose to the map.
+
+    Parameters
+    ----------
+    position
+        Specify the location of the magnetic rose on a map. See
+        :class:`pygmt.params.Position` for details.
+    width
+        Width of the rose in plot coordinates (append **i** (inch), **cm**
+        (centimeters), or **p** (points)), or append % for a size in percentage of map
+        width [Default is 15 %].
+    labels
+        A sequence of four strings to label the cardinal points W,E,S,N. Use an empty
+        string to skip a specific label. If the north label is ``"*"``, then a north
+        star is plotted instead of the north label. If set to ``True``, use the default
+        labels ``["W", "E", "S", "N"]``.
+    outer_pen
+        Draw the outer circle of the magnetic rose, using the given pen attributes.
+    inner_pen
+        Draw the inner circle of the magnetic rose, using the given pen attributes.
+    declination
+        Magnetic declination in degrees. By default, only a geographic north is plotted.
+        With this parameter set, a magnetic north is also plotted. A magnetic compass
+        needle is drawn inside the rose to indicate the direction to magnetic north.
+    declination_label
+        Label for the magnetic compass needle. Default is to format a label based on
+        **declination**. To bypass the label, set to ``"-"``.
+    intervals
+        Specify the annotation and tick intervals for the geographic and magnetic
+        directions. It can be a seqeunce of three or six values. If three values are
+        given, they are used for both geographic and magnetic directions. If six values
+        are given, the first three are used for geographic directions and the last three
+        for magnetic directions. Default is ``(30, 5, 1)``. **Note**: If
+        :gmt-term:`MAP_EMBELLISHMENT_MODE` is ``"auto"`` and the compass size is smaller
+        than 2.5 cm then the interval defaults are reset to ``(90,30, 3, 45, 15, 3)``.
+    box
+        Draw a background box behind the magnetic rose. If set to ``True``, a simple
+        rectangular box is drawn using :gmt-term:`MAP_FRAME_PEN`. To customize the box
+        appearance, pass a :class:`pygmt.params.Box` object to control style, fill, pen,
+        and other box properties.
+    $perspective
+    $verbose
+    $transparency
```

Comment:
Currently the docs look not correct with the leading "$":
<img width="171" height="86" alt="alias_parameter" src="https://github.com/user-attachments/assets/46fce505-1af7-408b-ab96-59d1ba1f9910" />